packages/databricks-vscode/src/language/notebooks/NotebookAccessVerifier.ts

```diff
@@ -0,0 +1,123 @@
+import {window} from "vscode";
+import {FeatureManager} from "../../feature-manager/FeatureManager";
+import {MultiStepAccessVerifier} from "../../feature-manager/MultiStepAccessVerfier";
+import {MsPythonExtensionWrapper} from "../MsPythonExtensionWrapper";
+import {WorkspaceStateManager} from "../../vscode-objs/WorkspaceState";
+
+export class NotebookAccessVerifier extends MultiStepAccessVerifier {
+    constructor(
+        private readonly featureManager: FeatureManager,
+        private readonly pythonExtension: MsPythonExtensionWrapper,
+        private readonly workspaceState: WorkspaceStateManager
+    ) {
+        super(["isPythonSdkInstalled", "isDbConnectEnabled"]);
+
+        this.disposables.push(
+            this.featureManager.onDidChangeState(
+                "debugging.dbconnect",
+                this.isDbConnectEnabled,
+                this
+            ),
+            this.pythonExtension.onDidChangePythonExecutable(
+                this.isPythonSdkInstalled,
+                this
+            )
+        );
+    }
+    async showSdkInstallPrompt() {
+        const mainMessagePart =
+            "Databricks SDK for python enables you to run databricks notebooks locally. Would you like to install it in the";
+
+        const env = await this.pythonExtension.pythonEnvironment;
+        const envMessagePart = env?.environment?.name
+            ? `environment ${env.environment.name}`
+            : `current environment`;
+
+        const sdkDetails =
+            await this.pythonExtension.getPackageDetailsFromEnvironment(
+                "databricks-sdk"
+            );
+        const sdkUpdateMessagePart = sdkDetails
+            ? `(databricks-sdk will be updated to the latest version: ${
+                  sdkDetails.version
+              } -> ${await this.pythonExtension.getLatestPackageVersion(
+                  "databricks-sdk"
+              )})`
+            : "";
+
+        const message = `${mainMessagePart} ${envMessagePart}? ${sdkUpdateMessagePart}`;
+        const result = await window.showInformationMessage(
+            message,
+            "Install",
+            "Change environment",
```

Comment:
After I clicked change environment I was not asked for installing the SDK again. I had to re-open the notebook.